### PR TITLE
irmin-pack: add lower_root config for lower layer

### PR DESCRIPTION
**Pull request by metanivek**
This PR adds the config key for the upcoming lower layer, with validation during `Repo.v`.

I didn't want to add a new error message just for an invalid `lower_root` value, and I also wanted good log message if the error occurs. To accomplish both, I am reusing `No_such_file_or_directory` and have added a string value for the file path. I think this also improves this error generally.

### CODE REVIEW

File: test/irmin-pack/common.ml [hunk 1]
```diff
@@ -179,6 +179,33 @@ module Alcotest = struct
...
+            match Astring.String.is_prefix ~affix:"Alcotest" exn_str with
+            (* Unfortunately we cannot access Alcotest's internal [Check_error]
+               so we do a dirty filter on its string representation to re-raise
+               instead of triggering another [Alcotest.failf]. *)
+            | true -> raise exn
+            | false ->
+                Alcotest.failf
+                  "Fail %s: expected function to raise Pack_error, but it \
+                   raised %s instead"
+                  msg exn_str))
```
Comment (Firobe) [inline test/irmin-pack/common.ml:207]: How about something like this instead with `Lwt.try_bind`? So we don't have to catch alcotest's exceptions
```suggestion
Lwt.try_bind (f x)
  (fun _ -> Alcotest.failf "Fail %s: expected function to raise, but it returned instead." msg)
  (function
    | Irmin_pack_unix.Errors.Pack_error e as exn -> (
            match pass e with
            | true -> Lwt.return_unit
            | false ->
                Alcotest.failf
                  "Fail %s: function raised unexpected exception %s" msg
                  (Printexc.to_string exn))
    | exn -> Alcotest.failf "Fail %s: expected function to raise Pack_error, but it \
                   raised %s instead" msg (Printexc.to_string exn))
```

**Review by Firobe — CHANGES_REQUESTED** [review]
Apart my only comment, the changes looks good to little me

File: src/irmin-pack/unix/errors.ml [hunk 2]
```diff
@@ -38,7 +38,7 @@ type base_error =
   [ `Double_close
   | `File_exists of string
   | `Invalid_parent_directory
-  | `No_such_file_or_directory
+  | `No_such_file_or_directory of string
```
Comment (art-w) [inline src/irmin-pack/unix/errors.ml:41]: This is great, thank you!

**Review by art-w — APPROVED** [review]

File: src/irmin-pack/unix/errors.ml (hunk 2 above)
Comment (metanivek) [reply src/irmin-pack/unix/errors.ml:41]: I'd like to take a look at all of our errors to decide which ones really deserve some extra context info. But, for a future PR. :smile:

**Pull request merged**